openstack/image/api.go

```diff
@@ -201,9 +214,10 @@ type APIConfig struct {
 // information needed to implement the image endpoints.
 type Service interface {
 	CreateImage(CreateImageRequest) (DefaultResponse, error)
-	UploadImage(string, io.Reader) (UploadImageResponse, error)
+	UploadImage(string, io.Reader) (NoContentImageResponse, error)
 	ListImages() ([]DefaultResponse, error)
```

Comment:
I know this isn't your bug - but I see that UploadImage also returns a response, when it should also return no response as indicated by the api spec.
http://developer.openstack.org/api-ref/image/v2/index.html?expanded=delete-an-image-detail,upload-binary-image-data-detail#upload-binary-image-data